.github/workflows/ci.yml

```diff
@@ -57,5 +58,20 @@ jobs:
               run: |
                   pip install -U -e .[tests]
 
+            - name: Set jupyter token env
+              run: echo "JUPYTER_TOKEN=$(openssl rand -hex 32)" >> $GITHUB_ENV
+
+            - name: Install Firefox geckodriver
+              uses: browser-actions/setup-firefox@latest
+              with:
+                  firefox-version: ${{ matrix.firefox }}
+
+            - name: Install geckodriver
```

Comment:
This installation (and the above) should only happen when matrix.browser==Firefox.